lib/plexy/logger.ex

```diff
@@ -97,16 +97,31 @@ defmodule Plexy.Logger do
   end
 
   defp metric_name(metric, name) do
-    app = Plexy.Config.get(:plexy, :app_name, System.get_env("APP_NAME") || "plexy")
     name = to_string(name)
-    "#{name}##{app}.#{metric}"
+    "#{name}##{app_name()}.#{metric}"
+  end
+
+  defp app_name do
+    Plexy.Config.get(:plexy, :app_name, System.get_env("APP_NAME") || "plexy")
```

Comment:
I changed this recently to this format b/c I didn't know how many apps using Plexy had the app_name set. Setting this now seems like a requirement. If we go forward with this PR we should probably require that the `app_name` is set via the plexy config. Maybe something like this but with a better error?

```elixir
Plexy.Config.get(:plexy, :app_name) || raise "you must set app_name for plexy config"
```

Reply:
I'm +1 for making it a requirement

Reply:
I updated the PR to reflect this. Struggling a bit on how to test the `raise` case. I'll have to think about it some more when not juggling other things.

Also, I wonder if this is a breaking/non-compatible change that should be a major version bump... 🤔

Reply:
`assert_raise\3` should work: https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#assert_raise/3

Reply:
Agreed that's what I'll use, I just have to be able to set/unset the config value at run time. That's the part I haven't figured out yet